src/components/ViewData.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import DataService from "../service/DataService";
+import { Descriptions } from "antd";
+import moment from "moment";
+import Layout, { Content, Footer, Header } from "antd/lib/layout/layout";
+import "../App.css";
+
+function ViewData() {
+  const { id } = useParams();
+
+  const [name, setName] = useState("");
```

Comment:
No need to use multiple state for multiple keys , use only one state and add data into it